core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1892,8 +1892,9 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
     }
   }
 
+  // Topic IDs are used with all self-managed quorum clusters and ZK cluster with IBP greater than or equal to 2.8
   def usesTopicId: Boolean =
-    interBrokerProtocolVersion >= KAFKA_2_8_IV0
+    usesSelfManagedQuorum || (requiresZookeeper && interBrokerProtocolVersion >= KAFKA_2_8_IV0)
```

Comment:
`requiresZookeeper` is always the inverse of `usesSelfManagedQuorum`, so this can simply be `usesSelfManagedQuorum || interBrokerProtocolVersion >= KAFKA_2_8_IV0`